aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -513,6 +513,13 @@ std::vector<Dimname> compute_baddbmm_outnames(
   return baddbmm_names;
 }
 
+bool are_names_equal(TensorImpl* self, TensorImpl* other) {
+  if (!impl::has_names(self) && !impl::has_names(other)) {
```

Comment:
nit: I don't know if this is better since it might be less clear to read, but I'd be tempted to put:
```
if (!impl::has_names(self)) {
  return !impl::has_names(other);
}
```
short-circuits the case where they're not equal and only the second arg has names.

Reply:
That's not correct. If `other` doesn't have names and `self` does have names, their names can still be equal if all of `self`'s names are None.